### PR TITLE
[wasm] Disable threading tests in System.IO

### PR DESCRIPTION
**Pull request by akoeplinger**
This allows the test suite to finish on WASM: `Tests run: 883, Errors: 0, Failures: 0, Skipped: 30. Time: 58.57082s`

### CODE REVIEW

**Review by mdh1418 — APPROVED** [review]
Changes look good to me. What were the 8 failing tests? Skips were added to failing tests in `System.Threading.Timer/tests` for helix to work, but do we not need to do the same for other suites now? There were several failures from #38355 that only appeared after disabling other failing tests in ` src/libraries/System.Threading.Tasks/tests/Task/TaskFromAsyncTest2.cs ` and `src/libraries/System.Threading.Tasks/tests/Task/TaskWaitAllAnyTest.cs`.

**Comment by akoeplinger** [comment]
@mdh1418 the failures were all due to `System.TypeInitializationException : The type initializer for 'System.Net.Sockets.SocketAsyncEngine' threw an exception.\n---- System.Net.InternalException : Exception of type 'System.Net.InternalException' was thrown`

Those tests are using NamedPipeServerStream which opens a Socket. We can indeed disable these.

**Review by steveisok — APPROVED** [review]

File: src/libraries/System.IO/tests/BufferedStream/BufferedStreamTests.cs [hunk 1]
```diff
@@ -17,7 +18,7 @@ protected override Stream CreateStream()
             return new BufferedStream(new MemoryStream());
         }
 
-        [Fact]
+        [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsThreadingSupported))]
```
Comment (stephentoub) [inline src/libraries/System.IO/tests/BufferedStream/BufferedStreamTests.cs:21]: What were the failures here?

Comment (akoeplinger) [reply src/libraries/System.IO/tests/BufferedStream/BufferedStreamTests.cs:21]: WASM-ERR: Unhandled Exception:
        WASM-ERR: System.Threading.SynchronizationLockException: Cannot wait on events on this runtime.
        WASM-ERR:    at System.Threading.WaitHandle.WaitOneCore(IntPtr waitHandle, Int32 millisecondsTimeout)
        WASM-ERR:    at System.Threading.WaitHandle.WaitOneNoCheck(Int32 millisecondsTimeout)
        WASM-ERR:    at System.Threading.WaitHandle.WaitOne()
        WASM-ERR:    at Microsoft.DotNet.XHarness.TestRunners.Xunit.ThreadlessXunitTestRunner.Run(String assemblyFileName, Boolean printXml, XunitFilters filters)
        WASM-ERR:    at Microsoft.DotNet.XHarness.TestRunners.Xunit.WasmApplicationEntryPoint.Run()
        WASM-ERR:    at SimpleWasmTestRunner.Main(String[] args)

Comment (stephentoub) [reply src/libraries/System.IO/tests/BufferedStream/BufferedStreamTests.cs:21]: Does that mean every async test is going to need to be disabled?  Or am I misunderstanding what this stack represents?

Comment (steveisok) [reply src/libraries/System.IO/tests/BufferedStream/BufferedStreamTests.cs:21]: It definitely seems that way.  Unless you know of any creative ideas to get a message pump in w/o significantly modifying the tests?

Comment (akoeplinger) [reply src/libraries/System.IO/tests/BufferedStream/BufferedStreamTests.cs:21]: Yes that's my current understanding. We're looking at ways how to fix that.

**Pull request merged**